hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java

```diff
@@ -166,47 +165,49 @@ protected boolean preDelete() {
 	}
 
 	protected void postDelete() {
-		final EventListenerGroup<PostDeleteEventListener> listenerGroup = listenerGroup( EventType.POST_DELETE );
-		if ( listenerGroup.isEmpty() ) {
-			return;
-		}
-		final PostDeleteEvent event = new PostDeleteEvent(
+		getFastSessionServices()
+				.eventListenerGroup_POST_DELETE
+				.fireLazyEventOnEachListener( this::newPostDeleteEvent, PostDeleteEventListener::onPostDelete );
+	}
+
+	PostDeleteEvent newPostDeleteEvent() {
+		return new PostDeleteEvent(
 				getInstance(),
 				getId(),
 				state,
 				getPersister(),
 				eventSource()
 		);
-		for ( PostDeleteEventListener listener : listenerGroup.listeners() ) {
+	}
+
+	protected void postCommitDelete(boolean success) {
+		final EventListenerGroup<PostDeleteEventListener> eventListeners = getFastSessionServices()
+				.eventListenerGroup_POST_COMMIT_DELETE;
+		if (success) {
+			eventListeners.fireLazyEventOnEachListener( this::newPostDeleteEvent, EntityDeleteAction::postCommitDeleteOnSuccess );
+		}
+		else {
+			eventListeners.fireLazyEventOnEachListener( this::newPostDeleteEvent, EntityDeleteAction::postCommitDeleteOnUnsuccessful );
+		}
+	}
+
+	private static void postCommitDeleteOnSuccess(PostDeleteEventListener listener, PostDeleteEvent event) {
+		if ( listener instanceof PostCommitDeleteEventListener ) {
+			listener.onPostDelete( event );
```

Comment:
Missing cast?

Reply:
hum yes that is looking odd. I suspect I can just remove the `if` - need to look better.

Reply:
I think it should be something like
```
if ( listener instanceof PostCommitDeleteEventListener ) {
if ( success ) {

					listener.onPostDelete( event );
				}
				else {
					((PostCommitDeleteEventListener) listener).onPostDeleteCommitFailed( event );
				}
}
```

Reply:
I'll do the same I did for `PostInsertEventListener`

Reply:
pushed an additional commit